executor/infoschema_reader.go

```diff
@@ -1118,3 +1121,25 @@ func dataForAnalyzeStatusHelper(sctx sessionctx.Context) (rows [][]types.Datum)
 func (e *memtableRetriever) setDataForAnalyzeStatus(sctx sessionctx.Context) {
 	e.rows = dataForAnalyzeStatusHelper(sctx)
 }
+
+func dataForServersInfo() ([][]types.Datum, error) {
```

Comment:
it's better to bind this function to `memtableRetriever`, like `setDataFromSessionVar()`.

Reply:
Done!